### PR TITLE
Keep track of source position for p4-14 switch statement labels

### LINKED ISSUE #3140: Error in compiling P4 14 programs
I am trying to compile some P4 programs (https://bitbucket.org/lavanyaj/switch-compiler/src/master/mapper/p4_programs/) using the P4C compiler. I am using the following command as mentioned in the P4C readme. 

`p4c --target bmv2 --arch v1model --std p4-14 my-p4-14-prog.p4
`
However, the compilation fails and the following error message is shown 

```
$ p4c --target bmv2 --arch v1model --std p4-14 l2l3simple.p4
actions.p4(32): [--Wwarn=shadow] warning: 'ipv6_prefix' shadows '.ipv6_prefix'
action set_ipv6_prefix_ucast(ipv6_prefix){
                             ^^^^^^^^^^^
tables.p4(90)
table ipv6_prefix {
^
actions.p4(36): [--Wwarn=shadow] warning: 'ipv6_prefix' shadows '.ipv6_prefix'
action set_ipv6_prefix_xcast(ipv6_prefix){
                             ^^^^^^^^^^^
tables.p4(90)
table ipv6_prefix {
^
actions.p4(28): [--Wwarn=shadow] warning: 'vrf' shadows '.vrf'
action set_vrf(vrf) {
               ^^^
tables.p4(59)
table vrf {
^
[--Werror=type-error] error: on_hit is not a legal label (action name)
[--Werror=type-error] error: on_miss is not a legal label (action name)

```

Is the conversion mechanism of the P4C compiler working properly or I am missing something.

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>
Fixes #3140

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
@ChrisDodd this reveals a bug in the v1 IR representation handling, where the source position information is actually lost. We should not really be using NameMap, but a map whose keys are ID, not cstrings.

File: frontends/p4/fromv1.0/converters.cpp [hunk 1]
```diff
@@ -504,7 +507,7 @@ class FixupExtern : public Modifier {
             type->annotations = type->annotations->addAnnotationIfNew(
                 IR::Annotation::nameAnnotation, new IR::StringLiteral(type->name.name), false);
             type->name = extname; }
-        // FIXME -- should create ctors based on attributes?  For now just create a
+        // FIXME -- should create ctors based on attributes?  For now j create a
```
Comment (rst0git) [inline frontends/p4/fromv1.0/converters.cpp:510]: Why `just` has been changed to `j` here?

File: ir/v1.def [hunk 2]
```diff
@@ -159,7 +159,12 @@ class NamedCond : If {
 
 class Apply : Expression {
     optional ID                         name;
+    // We should not use a NameMap, since it does not have source position information.
     inline NameMap<Vector<Expression>, ordered_map>  actions = {};
+    // We make up for it by storing the position information here.
+    inline NameMap<Path> position = {};
+    /*
+    // This is actually broken, since the srcInfo is 0 for all such vectors
```
Comment (rst0git) [inline ir/v1.def:167]: If it is broken maybe we should remove it instead of keeping it as commented-out code?

**Review by rst0git — APPROVED** [review]
I can confirm that the fix works as expected:

```console
l2l3simple.p4(43): [--Werror=type-error] error: on_hit is not a legal label (action name)
                on_hit {
                ^^^^^^
l2l3simple.p4(47): [--Werror=type-error] error: on_miss is not a legal label (action name)
                on_miss {
                ^^^^^^^
```

**Pull request merged**